pkgs/development/python-modules/entrypoint2/default.nix

```
@@ -0,0 +1,29 @@
+{ stdenv, buildPythonPackage, fetchPypi, decorator, EasyProcess, nose, pathpy }:
+
+buildPythonPackage rec {
+  pname = "entrypoint2";
+  version = "0.0.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1vpqqc4nff6658nyvzz3hr0mvdwr97x1wbvndjn78n54y91s4wms";
+  };
+
+  postPatch = ''
+    substituteInPlace requirements.txt --replace "argparse" ""
+    substituteInPlace ./entrypoint2/__init__.py \
+      --replace "except UsageError, e:" "except UsageError as e:"
+  '';
+
+  propagatedBuildInputs = [ decorator ];
+  checkInputs = [ EasyProcess nose pathpy ];
+
+  checkPhase = "nosetests";
```

```suggestion
  checkPhase = ''
    runPhase preCheck
    nosetests
    runPhase postCheck
  '';
```

So that we don't lose the pre and post phases :)

---

FYI, it is called `runHook` instead of `runPhase` :)